src/shogun/features/DenseFeatures.cpp

```diff
@@ -265,23 +267,51 @@ template<class ST> void CDenseFeatures<ST>::feature_subset(int32_t* idx, int32_t
 	}
 }
 
-template<class ST> SGMatrix<ST> CDenseFeatures<ST>::get_feature_matrix()
+template <class ST>
+SGMatrix<ST> CDenseFeatures<ST>::get_feature_matrix()
 {
 	if (!m_subset_stack->has_subsets())
 		return feature_matrix;
 
-	SGMatrix<ST> submatrix(num_features, get_num_vectors());
+	SGMatrix<ST> target(num_features, get_num_vectors());
+	copy_feature_matrix(target);
+	return target;
+}
+
+template <class ST>
+void CDenseFeatures<ST>::copy_feature_matrix(SGMatrix<ST> target, index_t column_offset) const
+{
+	REQUIRE(column_offset>=0, "Column offset (%d) cannot be negative!\n", column_offset);
 
-	/* copy a subset vector wise */
-	for (int32_t i=0; i<submatrix.num_cols; ++i)
+	index_t num_vecs=get_num_vectors();
+	index_t num_cols=num_vecs+column_offset;
+
+	REQUIRE(target.matrix!=nullptr, "Provided matrix is not allocated!\n");
+	REQUIRE(target.num_rows==num_features,
+			"Number of rows (%d for given matrix) should be equal to the number of features (%d)!\n",
```

Comment:
`Number of rows of given matrix (%d)` would be more inline the messages we have. But that is very very minor